pkg/controller/namespace/namespace_controller.go

```diff
@@ -209,25 +209,61 @@ func deleteAllContent(kubeClient client.Interface, experimentalMode bool, namesp
 	return estimate, nil
 }
 
-// syncNamespace makes namespace life-cycle decisions
-func syncNamespace(kubeClient client.Interface, experimentalMode bool, namespace api.Namespace) (err error) {
-	if namespace.DeletionTimestamp == nil {
-		return nil
+// updateNamespaceFunc is a function that makes an update to a namespace
+type updateNamespaceFunc func(kubeClient client.Interface, namespace *api.Namespace) (*api.Namespace, error)
+
+// retryOnConflictError retries the specified delegate if there was a conflict error
+func retryOnConflictError(kubeClient client.Interface, namespace *api.Namespace, delegate updateNamespaceFunc) (result *api.Namespace, err error) {
+	result = namespace
+	retry := true
+	latestNamespace := namespace
+	for retry {
+		result, err = delegate(kubeClient, latestNamespace)
+		if err == nil {
+			retry = false
```

Comment:
See my other about returning inside the loop, but yes

On Sep 21, 2015, at 5:13 PM, Derek Carr <notifications@github.com> wrote:

In pkg/controller/namespace/namespace_controller.go
<https://github.com/kubernetes/kubernetes/pull/14297#discussion_r40026726>:

> -// syncNamespace makes namespace life-cycle decisions
> -func syncNamespace(kubeClient client.Interface, experimentalMode bool, namespace api.Namespace) (err error) {
> -	if namespace.DeletionTimestamp == nil {
> -		return nil
> +// updateNamespaceFunc is a function that makes an update to a namespace
> +type updateNamespaceFunc func(kubeClient client.Interface, namespace *api.Namespace) (*api.Namespace, error)
> +
> +// retryOnConflictError retries the specified delegate if there was a conflict error
> +func retryOnConflictError(kubeClient client.Interface, namespace *api.Namespace, delegate updateNamespaceFunc) (result *api.Namespace, err error) {
> +	result = namespace
> +	retry := true
> +	latestNamespace := namespace
> +	for retry {
> +		result, err = delegate(kubeClient, latestNamespace)
> +		if err == nil {
> +			retry = false

assume you mean break

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/14297/files#r40026726>.